jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/component/ContainerLifeCycle.java

```diff
@@ -882,7 +882,7 @@ public void updateBeans(final Collection<?> oldBeans, final Collection<?> newBea
     @Override
     public <T> Collection<T> getContainedBeans(Class<T> clazz)
     {
-        Set<T> beans = new HashSet<>();
+        Set<T> beans = new LinkedHashSet<>();
```

Comment:
The javadoc of this method mandates an order, which is not respected when using a `HashSet`.

Furthermore, it may be a problem when bean A contains bean B, and then you operate on the resulting `Collection`: bean B may expect that since bean A is an ancestor, the operation has already happened for A, which may not be the case if the `HashSet` reorders the beans randomly.

Reply:
If this bug has been lurking for so long, this means we're lacking coverage. Could you please add a test that asserts this ordering?